### ISSUE TITLE
android does not have pwd.getpwall()

### ISSUE THREAD

**Issue by 490c593f-f636-409f-bb35-6abeb38a4595** [python/cpython#71043]
BPO | [26856](https://bugs.python.org/issue26856)
--- | :---
Nosy | @xdegaye, @Fak3, @moreati
PRs | <li>python/cpython#4561</li>
Dependencies | <li>bpo-27027: add the &#x27;is_android&#x27; attribute to test.support</li><li>bpo-32033: The pwd module implementation incorrectly sets some attributes to None</li>
Files | <li>[pwd.patch](https://bugs.python.org/file42608/pwd.patch "Uploaded as text/plain at 2016-04-26.13:29:04 by @xdegaye")</li><li>[pwd_2.patch](https://bugs.python.org/file42913/pwd_2.patch "Uploaded as text/plain at 2016-05-20.09:22:48 by @xdegaye")</li><li>[pwd_3.patch](https://bugs.python.org/file45855/pwd_3.patch "Uploaded as text/plain at 2016-12-12.09:22:06 by @xdegaye")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/xdegaye'
closed_at = None
created_at = <Date 2016-04-26.13:29:05.170>
labels = ['3.7', 'tests', 'type-bug', 'library', 'build']
title = 'android does not have pwd.getpwall()'
updated_at = <Date 2017-11-25.16:33:07.817>
user = 'https://github.com/xdegaye'
```

bugs.python.org fields:
```python
activity = <Date 2017-11-25.16:33:07.817>
actor = 'xdegaye'
assignee = 'xdegaye'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)', 'Tests', 'Cross-Build']
creation = <Date 2016-04-26.13:29:05.170>
creator = 'xdegaye'
dependencies = ['27027', '32033']
files = ['42608', '42913', '45855']
hgrepos = []
issue_num = 26856
keywords = ['patch']
message_count = 6.0
messages = ['264279', '265919', '282975', '283082', '306962', '306966']
nosy_count = 4.0
nosy_names = ['xdegaye', 'python-dev', 'Roman.Evstifeev', 'Alex.Willmer']
pr_nums = ['4561']
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue26856'
versions = ['Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by 490c593f-f636-409f-bb35-6abeb38a4595** [comment]
The pwd module is broken when a member of the 'passwd’ data structure is NULL (bpo-32033).
Remove \_getpwall() and therefore skip test_pwd on Android until bpo-32033 is fixed.

**Comment by 490c593f-f636-409f-bb35-6abeb38a4595** [comment]
New changeset 76fdac4c9f53eb8433a54bd3daf9f5cc2e702a44 by xdegaye in branch 'master':
bpo-26856: Skip test_pwd on Android until bpo-32033 is fixed (GH-4561)
https://github.com/python/cpython/commit/76fdac4c9f53eb8433a54bd3daf9f5cc2e702a44

**Comment by kumaraditya303** [comment]
Fixed by #4561
